src/throttle/amoc_throttle_process.erl

```diff
@@ -24,117 +16,79 @@
+-define(DEFAULT_MSG_TIMEOUT, 60000). %% one minute
 
--record(state, {can_run_fn = true :: boolean(),
-                pause = false :: boolean(),
-                max_n :: non_neg_integer(),
-                name :: atom(),
-                n :: integer(),
-                interval = 0 :: amoc_throttle:interval(),  %%ms
-                delay_between_executions = 0 :: non_neg_integer(),  %%ms
+-record(state, {name :: atom(),
+                delay_between_executions :: timeout(),  %% ms
+                max_n :: infinity | non_neg_integer(),
+                n = 0 :: non_neg_integer(),
+                can_run_fn = true :: boolean(),
                 tref :: timer:tref() | undefined,
                 schedule = [] :: [AmocThrottleRunnerProcess :: pid()],
-                schedule_reversed = [] :: [AmocThrottleRunnerProcess :: pid()]}).
+                schedule_reversed = [] :: [AmocThrottleRunnerProcess :: pid()]
+               }).
 
 -type state() :: #state{}.
 %%------------------------------------------------------------------------------
 %% Exported functions
 %%------------------------------------------------------------------------------
 
--spec start_link(atom(), amoc_throttle:interval(), amoc_throttle:rate()) -> {ok, pid()}.
-start_link(Name, Interval, Rate) ->
-    gen_server:start_link(?MODULE, {Name, Interval, Rate}, []).
-
--spec stop(pid()) -> ok.
-stop(Pid) ->
-    gen_server:cast(Pid, stop_process).
+-spec start_link(atom(), amoc_throttle:rate(), timeout()) -> gen_server:start_ret().
+start_link(Name, MaxN, Delay) ->
+    gen_server:start_link(?MODULE, {Name, MaxN, Delay}, []).
 
 -spec run(pid(), pid()) -> ok.
 run(Pid, RunnerPid) ->
     gen_server:cast(Pid, {schedule, RunnerPid}).
 
--spec update(pid(), amoc_throttle:interval(), amoc_throttle:rate()) -> ok.
-update(Pid, Interval, Rate) ->
-    gen_server:cast(Pid, {update, Interval, Rate}).
-
--spec pause(pid()) -> ok.
-pause(Pid) ->
-    gen_server:cast(Pid, pause_process).
-
--spec resume(pid()) -> ok.
-resume(Pid) ->
-    gen_server:cast(Pid, resume_process).
-
--spec get_state(pid()) -> map().
-get_state(Pid) ->
-    gen_server:call(Pid, get_state).
-
--spec get_throttle_process(amoc_throttle:name()) ->
-    {error, no_throttle_process_registered} | {ok, pid()}.
-get_throttle_process(Name) ->
-    case pg:get_members(?PG_SCOPE, Name) of
-        [] ->
-            {error, no_throttle_process_registered};
-        List -> %% nonempty list
-            N = rand:uniform(length(List)),
-            {ok, lists:nth(N, List)}
-    end.
-
--spec get_throttle_processes(amoc_throttle:name()) ->
-    {error, no_throttle_process_registered} | {ok, [pid()]}.
-get_throttle_processes(Name) ->
-    case pg:get_members(?PG_SCOPE, Name) of
-        [] ->
-            {error, no_throttle_process_registered};
-        List ->
-            {ok, List}
-    end.
+-spec update(pid(), amoc_throttle:rate(), timeout()) -> ok.
+update(Pid, MaxN, Delay) ->
```

Comment:
please add a comment that setting the delay to infinity results in effective pausing of the throttle process.